### ISSUE TITLE
[ENH] address duplications in serialization methods of deep learning base class, `BaseDeepClassifier.save` and `BaseDeepRegressor.save`

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#6137]
There is substantial duplication in the serialization methods of deep learning base classes, `BaseDeepClassifier.save` and `BaseDeepRegressor.save`, and corresponding loader methods.

The methods are are almost identical, the main difference comes from only `BaseDeepClassifier.save` having `cloudpickle` support.

Without this difference, the DRY refactor strategy would be moving identical methods to a joint base class or mixin, similar to `BasePanelMixin`.

However, in the current situation there is some matching to do, possibly adding `cloudpickle` support to serialization of regressors too.

By naive inspection, I would expect that the classifier methods can be copied over, and at most some docstring changes (reference to classifier etc) are necessary.

If done soon, this would interact with https://github.com/sktime/sktime/pull/6098, so contibutors should branch off https://github.com/sktime/sktime/pull/6098 and avoid working close to the 0.30.0 release (likely in May 2024).

**Comment by iaryangoyal** [comment]
@fkiraly may I start with this?

**Comment by fkiraly** [comment]
of course! At the moment, as said, I would advise to branch off #6098.

**Comment by iaryangoyal** [comment]
#6134 Please check the commit files in this

**Comment by iaryangoyal** [comment]
@fkiraly Please refer to the same and check I have branched off #6098 Please check once and let me know

**Comment by Utkarsh-Aggarwal** [comment]
i am currently working on this issue
